src/stream/src/task/barrier_manager/managed_state.rs

```diff
@@ -148,6 +155,32 @@ impl ManagedBarrierState {
     pub(crate) fn clear_all_states(&mut self) {
         self.epoch_barrier_state_map.clear();
         self.create_mview_progress.clear();
+        self.exited_actors.clear();
+    }
+
+    /// Notify unexpected actor exit with given `actor_id`.
+    pub(crate) fn notify_exit(&mut self, actor_id: ActorId) {
+        self.exited_actors.insert(actor_id);
+        for barrier_state in self.epoch_barrier_state_map.values_mut() {
+            #[allow(clippy::single_match)]
+            match barrier_state.inner {
+                ManagedBarrierStateInner::Issued {
+                    ref remaining_actors,
+                    ref mut collect_notifier,
+                } => {
+                    if remaining_actors.contains(&actor_id) && let Some(collect_notifier) = collect_notifier.take() && collect_notifier
+                            .send(Err(anyhow!(format!(
+                                "Actor {} exit unexpectedly",
+                                actor_id
+                            ))
```

Comment:
How about also encoding the original error reported by the actor in this message? So that the meta service will give a more reasonable error message.